src/server/req_delete.c

```diff
@@ -394,6 +394,7 @@ req_deletejob(struct batch_request *preq)
 				parent->ji_ajtrk->tkm_tbl[offset].trk_substate =
 					JOB_SUBSTATE_TERMINATED;
 				set_subjob_tblstate(pjob, offset, JOB_STATE_EXPIRED);
+				parent->ji_ajtrk->tkm_dsubjsct++;
```

Comment:
Should we also need to update the count for the condition (jt == IS_ARRAY_ArrayJob).